### PR TITLE
Support location correctly without test

### PR DESCRIPTION
**Pull request by hythof**
Relate: https://github.com/WOVNio/equalizer/issues/11477
Replace `Location` header correctly.

### CODE REVIEW

File: src/main/java/com/github/wovnio/wovnjava/Headers.java [hunk 1]
```diff
@@ -211,6 +211,62 @@ String redirectLocation(String lang) {
...
+            } else {
+                queryLangCode = "?wovn=" + lang;
+            }
+        } else if (settings.urlPattern.equals("subdomain")) {
+            subdomainLangCode = lang + ".";
+        } else {
+            pathLangCode = "/" + lang;
+        }
+        return locationProtocol + "://" + subdomainLangCode + host + pathLangCode + path + queryLangCode;
+    }
```
Comment (torkelbe) [inline src/main/java/com/github/wovnio/wovnjava/Headers.java]: What is the difference between
```
String redirectLocation(String lang) {...}
```
and
```
public String locationWithLangCode(String location) {...}
```
To me, they seem to try to do the same thing.

File: src/main/java/com/github/wovnio/wovnjava/Headers.java [hunk 2]
```diff
@@ -234,21 +290,33 @@ String removeLang(String uri, String lang) {
...
-                if (Pattern.compile("\\?").matcher(loc).find()) {
-                    res.setHeader("Location", loc + "?wovn=" + langCode());
-                } else {
-                    res.setHeader("Location", loc + "&wovn=" + langCode());
-                }
-            } else if (this.settings.urlPattern.equals("subdomain")) {
-                loc = loc.replaceFirst("//([^.]+)", "//" + langCode() + "\\.$1");
-                res.setHeader("Location", loc);
+    private String pathJoin(String left, String right) {
+        boolean l = left.endsWith("/");
```
[the remark this answers is not in the document]
Comment (hythof) [reply src/main/java/com/github/wovnio/wovnjava/Headers.java]: If left is "foo" and right is "./bar", the result is "foo/./bar".
After that it's converted "foo/./bar" to "foo/bar" by `pathNormalize`.

[the remark this answers is not in the document]
Comment (hythof) [reply src/main/java/com/github/wovnio/wovnjava/Headers.java]: If left is "foo" and right is "./bar", the result is "foo/./bar".
After that it's converted "foo/./bar" to "foo/bar" by `pathNormalize`.

File: src/main/java/com/github/wovnio/wovnjava/Headers.java (hunk 1 above)
Comment (hythof) [reply src/main/java/com/github/wovnio/wovnjava/Headers.java]: > redirectLocation
This method is to create new URL from browser URL and specific language code.
borwser URL: https://foo.com/
language code: fr
result: https://foo.com/fr/

> locationWithLangCode
This method is to create new URL from specific URL and current language code.
borwser URL: https://foo.com/fr/signin
specific URL: /home
result: https://foo.com/fr/home

Comment (torkelbe) [reply src/main/java/com/github/wovnio/wovnjava/Headers.java]: Gotcha. Thanks for the examples.

**Pull request merged**